src/options.rs

```diff
@@ -178,11 +178,25 @@ impl ResolvedRoundingOptions {
         options: DifferenceSettings,
         operation: DifferenceOperation,
         unit_group: UnitGroup,
+        dissallow_week_and_day: bool,
         fallback_largest: TemporalUnit,
         fallback_smallest: TemporalUnit,
     ) -> TemporalResult<Self> {
-        // 1. NOTE: The following steps read options and perform independent validation in alphabetical order.
-        // 2. Let largestUnit be ? GetTemporalUnitValuedOption(options, "largestUnit", unitGroup, auto).
+        // Week and day is the only use case where any of the units are not allowed.
+        if dissallow_week_and_day {
```

Comment:
thought: is there anything preventing moving this into `PlainYearMonth::diff` vs. where it's currently at in `ResolvedRoundingOptions::from_diff_settings`.

This only affects the one code path from `PlainYearMonth`, I think, so maybe this could work in `PlainYearMonth::diff`. This would then avoid the check occurring for the other callers.